packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -55,6 +57,11 @@ export class EngineReportingExtension<TContext = any>
     });
   }
 
+  // If GraphQL validation has started, then GraphQL parsing must have succeeded
+  public validationDidStart(): ((...errors: Error[]) => void) | void {
+    this.gqlParsingSucceeded = true;
```

Comment:
I don't really think you need an explicit bool for this (or parameter to addTrace). parsing failed IFF there is no AST.

That said, I do think that this is a fine place to set documentAST, instead of in executionDidStart.  (I mean, the best place would be in parsingDidEnd, but in the extension stack (but not the plugin!) API you can't do that. So there should be a comment saying that it should be in parsingDidEnd if we were on the plugin API.)

Reply:
I started to identify it by whether `documentAST` was defined, but figured that being explicit with a well-named variable (and not reliant on the request pipeline) would be better.

Reply:
How would I access the `documentAST` in the `validationDidEnd` from the parent context to set it here, though?